src/System.Runtime/tests/System/TimeZoneInfo.cs

```diff
@@ -1745,6 +1746,44 @@ public static void TestLisbonDaylightSavingsWithNoOffsetChange(string dateTimeSt
         }
     }
 
+    [Fact]
+    public static void TestGetSystemTimeZones()
+    {
+        ReadOnlyCollection<TimeZoneInfo> timeZones = TimeZoneInfo.GetSystemTimeZones();
+        Assert.NotEmpty(timeZones);
+        Assert.Contains(timeZones, t => t.Id == s_strPacific);
+        Assert.Contains(timeZones, t => t.Id == s_strSydney);
+        Assert.Contains(timeZones, t => t.Id == s_strGMT);
+        Assert.Contains(timeZones, t => t.Id == s_strTonga);
+        Assert.Contains(timeZones, t => t.Id == s_strBrasil);
+        Assert.Contains(timeZones, t => t.Id == s_strPerth);
+        Assert.Contains(timeZones, t => t.Id == s_strBrasilia);
+        Assert.Contains(timeZones, t => t.Id == s_strNairobi);
+        Assert.Contains(timeZones, t => t.Id == s_strAmsterdam);
+        Assert.Contains(timeZones, t => t.Id == s_strRussian);
+        Assert.Contains(timeZones, t => t.Id == s_strLibya);
+        Assert.Contains(timeZones, t => t.Id == s_strCatamarca);
+        Assert.Contains(timeZones, t => t.Id == s_strLisbon);
+        Assert.Contains(timeZones, t => t.Id == s_strNewfoundland);
+
+        // ensure the TimeZoneInfos are sorted by BaseUtcOffset and then DisplayName.
+        TimeZoneInfo previous = timeZones[0];
+        for (int i = 1; i< timeZones.Count; i++)
+        {
+            TimeZoneInfo current = timeZones[i];
+            int baseOffsetsCompared = current.BaseUtcOffset.CompareTo(previous.BaseUtcOffset);
```

Comment:
I think the documentation is incorrect.  The code also has a comment that the TimeZones are sorted by DisplayName, but the TimeZoneInfoComparer sorts them by BaseUtcOffset, and then by DisplayName.
See https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/TimeZoneInfo.cs#L5479.